src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/IntellisenseQuickInfoBuilder.cs

```diff
@@ -79,16 +107,67 @@ internal static class IntellisenseQuickInfoBuilder
             }
 
             var content = new ContainerElement(
-                                ContainerElementStyle.Stacked,
+                                ContainerElementStyle.Stacked | ContainerElementStyle.VerticalPadding,
                                 elements);
 
             return new IntellisenseQuickInfoItem(trackingSpan, content);
         }
 
-        private static ClassifiedTextElement BuildClassifiedTextElement(QuickInfoSection section)
+        private static IEnumerable<object> BuildClassifiedTextElements(QuickInfoSection section)
         {
-            return new ClassifiedTextElement(section.TaggedParts.Select(
-                    part => new ClassifiedTextRun(part.Tag.ToClassificationTypeName(), part.Text)));
+            var paragraphs = new List<object>();
+            var currentParagraph = new List<object>();
+            var currentRuns = new List<ClassifiedTextRun>();
+            foreach (var part in section.TaggedParts)
+            {
+                if (part.Tag == TextTags.LineBreak)
+                {
+                    if (currentRuns.Count == 0)
+                    {
+                        if (currentParagraph.Count > 0)
+                        {
+                            if (currentParagraph.Count == 1)
+                            {
+                                paragraphs.Add(currentParagraph[0]);
```

Comment:
if it is first time, then put paragraph as it is.

Reply:
Now documented in the `CreateParagraphFromLines` method.